astropy/modeling/fitting.py

```diff
@@ -1591,6 +1601,25 @@ def _convert_input(x, y, z=None, n_models=1, model_set_axis=0):
     return farg
 
 
+def _filter_input(farg: tuple, filter_non_finite=False):
+    """Check for and filter non-finite inputs"""
+
+    if np.any([not np.all(np.isfinite(arg)) for arg in farg]) and not filter_non_finite:
+        warnings.warn("Non-finite input data has been detected, this may cause the fit to fail!\n",
+                      AstropyUserWarning)
+
+    if filter_non_finite:
+        finite_index = np.zeros(farg[0].shape, dtype=bool)
+        for arg in farg:
+            finite_index[np.logical_not(np.isfinite(arg))] = True
```

Comment:
I think a shorter version is easier to read, but that's personal preference...
```suggestion
            finite_index[~(np.isfinite(arg))] = True
```

Reply:
See 7fa1b2d.